airflow/providers/microsoft/psrp/hooks/psrp.py

```diff
@@ -16,103 +16,222 @@
 # specific language governing permissions and limitations
 # under the License.
 
-from time import sleep
+import re
+from contextlib import contextmanager
+from logging import DEBUG, ERROR, INFO, WARNING
+from typing import Any, Dict, Optional
+from weakref import WeakKeyDictionary
 
-from pypsrp.messages import ErrorRecord, InformationRecord, ProgressRecord
+from pypsrp.messages import MessageType
 from pypsrp.powershell import PowerShell, PSInvocationState, RunspacePool
 from pypsrp.wsman import WSMan
 
 from airflow.exceptions import AirflowException
 from airflow.hooks.base import BaseHook
 
+INFORMATIONAL_RECORD_LEVEL_MAP = {
+    MessageType.DEBUG_RECORD: DEBUG,
+    MessageType.ERROR_RECORD: ERROR,
+    MessageType.VERBOSE_RECORD: INFO,
+    MessageType.WARNING_RECORD: WARNING,
+}
+
 
 class PSRPHook(BaseHook):
```

Comment:
Hmm we’ve been recently renaming things in the AWS provider, maybe we should also take the change to rename this to `PsrpHook` to fit the naming converntion (operator as well).

Reply:
Or `PowerShellRemotingHook` – something like that might help users who don't know that this is what they should use.

Reply:
Fix merged into e9fe6995ffdcdd73e3d92c1c1ba2ddba8b832c5a.